Add ability for MeshVisual to be displayed in frames different from the map frame

JustusBraun opened the pull request:
Previously the Mesh would disappear when changing the fixed frame, now it can be displayed as expected.

amock commented:
Great!

I successfully tested it with mesh geometries published via a topic using https://github.com/amock/mesh_msgs_examples along with a static transform publisher. However, meshes loaded directly from a file still can't be transformed correctly.

That said, I'm fine with merging this as it is an improvement already.

JustusBraun commented:
I added the ability to transform MapDisplays, and you can also select the reference frame of a map loaded from file. Also I noticed that the map would not be retransformed when the transform between the fixed frame and the map frame changed e.g. map to odom transform. This is now also fixed and working

amock commented:
Works on my machine :D Ubuntu 24, ROS 2 jazzy

amock commented:
Works on my other machine. Ubuntu 22, ROS 2 humble (Laptop)

amock commented:
Works on my other machine: Ubuntu 22, ROS 2 humble (desktop PC)
